Tango skin :: hidpi fixes

ronso0 opened the pull request:
I fixed some major bugs appearing when Tango skin is scaled up (see [lp:1751290](https://bugs.launchpad.net/mixxx/+bug/1751290)), as well as some minor UX issues

- qss: split 'background' shorthand into 'background-color' and 'image' (otherwise those images stayed at 100%)
- Fx flow graphics (were completely gone, width of 'border-image' was not scaled)
- Effect selector (removed explicit 'size', added background to effect list toggle, #1518 also fixes the effects list display)
- Fx parameter row (rebuild, margin is not scaled but was necessary for previous stacked layout)
- beatsize spinboxes, buttons & spinbox in Library
- borders in sampler and preview deck

Screenshot 1920x1080, mixxx & system font at 200%
![tango-hidpi-x2_2018-02-27](https://user-images.githubusercontent.com/5934199/36694335-5f904a82-1b3e-11e8-8144-1810f97d409a.png)

Commit:
scaling FxUnits: fix fx flow graphics, parameter knobs, EffectSelector etc.

Commit:
fix beatsize spinboxes and looping section for scaling

Commit:
fix & simplify Artist/Title/Time templates, so that elapsed/remaining doesn't squeeze Artist anymore

Commit:
improve deck controls toggle: sits at the far left/right now. Add hover effect to expanded deck controls

Commit:
Vinyl controls: replace margin with fixed spacer to maintain layout padding when scaling the skin

Commit:
Preview deck: replace margin with fixed spacer

Commit:
qss: remove 'background' shorthand definitions, split them into 'background-color' and 'image' to allow proper scaling of background images

Commit:
Samplers: replace margin with fixed spacer to maintain layout padding when scaling the skin

Commit:
add comments, clean up

Commit:
remove Library toggle from topbar, rename controls

ronso0 commented:
Don't merge yet, I need to fix Artist/Time row and Overview.

Commit:
fix Time container to prevent squeezing of track overview

Commit:
fix QPushbutton text in Library

ronso0 commented:
Ready to test!

Commit:
add background to FX unit label, fix text position for collapsed & expanded state

Commit:
Sampler row: fix Cover/Play in expanded sampler, optimize row spacers, add bg to row expand toggle

Commit:
fix Artist/Title alignment for right-hand decks when using 'Symmetric Time/Duration. Fixes A. of lp1750902

Commit:
revert tooltip change for decks' Play button

ronso0 commented:
I just finished this PR.
Besides the scaling fixes, there are a few minor changes:
- FX unit labels and show_parameters toggle have a background now, so they visually relate more to the actual FX unit. Same applies to sampler row expand toggle
- toggle for additional, hidden deck controls (quantize, repeat etc.) now sits at the outside of deck buttons row
- right-hadn decks' Artist/Title are aligned right (fixes A. of [lp:1750902](https://bugs.launchpad.net/mixxx/+bug/1750902))
- I removed Library toggle from the top row. Didn't use it much, Alt+Tab is much quicker to hide tracks and playlists from snoopy fellows

Be-ing commented:
LGTM, thanks!

Be-ing commented on res/skins/Tango/fx_toggle_selector.xml:
```diff
@@ -51,10 +51,10 @@ Variables:
       </Template>
 
       <EffectSelector>
-        <Size>82f,20f</Size>
```
This broke the scaling with #1518. I'll revert it in that PR.

ronso0 commented:
it seems this was merged too early: I just discovered a bug with some button icons after I fixed those to scale well. All pause icons appear blurred, although at least their height matches the buttons' height.

Fixed in #1531
